open-ce/build_feedstock.py

```diff
@@ -147,42 +141,46 @@ def _set_local_src_dir(local_src_dir_arg, recipe, recipe_config_file):
         if 'LOCAL_SRC_DIR' in os.environ:
             del os.environ['LOCAL_SRC_DIR']
 
-def build_feedstock(args_string=None):
+def build_feedstock_from_command(command, # pylint: disable=too-many-arguments
+                                 recipe_config_file=None,
+                                 output_folder=utils.DEFAULT_OUTPUT_FOLDER,
+                                 extra_channels=None,
+                                 conda_build_config=utils.DEFAULT_CONDA_BUILD_CONFIG,
+                                 local_src_dir=None):
     '''
-    Entry function.
+    Build a feedstock from a build_command object.
     '''
-    parser = make_parser()
-    args = parser.parse_args(args_string)
-
+    if not extra_channels:
+        extra_channels = []
     saved_working_directory = None
-    if args.working_directory:
+    if command.repository:
         saved_working_directory = os.getcwd()
-        os.chdir(os.path.abspath(args.working_directory))
+        os.chdir(os.path.abspath(command.repository))
 
-    build_config_data, recipe_config_file  = load_package_config(args.recipe_config_file)
+    build_config_data, recipe_config_file  = load_package_config(recipe_config_file)
 
-    args.recipes = utils.parse_arg_list(args.recipe_list)
+    recipes_to_build = utils.parse_arg_list(command.recipe)
 
     # Build each recipe
     for recipe in build_config_data['recipes']:
-        if args.recipes and recipe['name'] not in args.recipes:
+        if recipes_to_build and recipe['name'] not in recipes_to_build:
             continue
 
         config = get_or_merge_config(None)
         config.skip_existing = True
-        config.output_folder = args.output_folder
-        config.variant_config_files = [args.conda_build_config]
+        config.output_folder = output_folder
+        config.variant_config_files = [conda_build_config]
 
         recipe_conda_build_config = os.path.join(os.getcwd(), "config", "conda_build_config.yaml")
         if os.path.exists(recipe_conda_build_config):
             config.variant_config_files.append(recipe_conda_build_config)
 
-        config.channel_urls = args.channels_list + build_config_data.get('channels', [])
+        config.channel_urls = command.channels + extra_channels + build_config_data.get('channels', [])
```

Comment:
I think we want extra_channels to come first, since it will have the local channel in it.